src/icalendar/cli.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+"""iCalendar utility"""
+from __future__ import unicode_literals
+
+import argparse
+import sys
+from datetime import datetime
+
+from . import Calendar, __version__
+
+
+_template = """Organiser: {organiser}
+Attendees:
+  {attendees}
+Summary: {summary}
+When: {time_from}-{time_to}
+Location: {location}
+Comment: {comment}
+Description:
+
+{description}
+
+"""
+
+
+def _format_name(address):
+    """Retrieve the e-mail and optionally the name from an address.
+
+    :arg vCalAddress address: An address object.
+
+    :returns str: The name and optionally the e-mail address.
+    """
+    if not address:
+        return ''
+
+    email = address.title().split(':')[1]
+    if 'cn' in address.params:
+        return '{} <{}>'.format(address.params['cn'], email)
+
+    return email
+
+
+def _format_attendees(attendees):
+    """Format the list of attendees.
+
+    :arg any attendees: Either a list, a string or a vCalAddress object.
+
+    :returns str: Formatted list of attendees.
+    """
+    if type(attendees) == list:
+        return '\n  '.join(map(_format_name, attendees))
+    return _format_name(attendees)
+
+
+def view(input_handle, output_handle):
+    """Make a human readable summary of an iCalendar file.
+
+    :arg stream handle: Open readable handle to an iCalendar file.
+
+    :returns str: Human readable summary.
+    """
+    cal = Calendar.from_ical(input_handle.read())
```

Comment:
I'd enclose the whole body in a ``with`` statement, so that the file handles are closed automatically at the end. Regarding the discussion here: https://stackoverflow.com/a/41495679/1337474 the following should work:

```
with input_handle as fih, output_handle as foh:
    ...
```

Reply:
I would rather keep this function free from side-effects, on the other hand, closing the file handles in `main()` will be messy...

Reply:
fair point